webhooks/pod_webhook.go

```diff
@@ -206,63 +224,85 @@ func (m *PodMutator) getFeatureFlag(ctx context.Context, name string, namespace
 	return ffConfig
 }
 
-func (m *PodMutator) injectSidecar(pod *corev1.Pod, configMap string, featureFlag *corev1alpha1.FeatureFlagConfiguration) ([]byte, error) {
+func (m *PodMutator) injectSidecar(pod *corev1.Pod, featureFlags []*corev1alpha1.FeatureFlagConfiguration) ([]byte, error) {
 	m.Log.V(1).Info(fmt.Sprintf("Creating sidecar for pod %s/%s", pod.Namespace, pod.Name))
 
 	commandSequence := []string{
-		"start", "--uri", "/etc/flagd/config.json",
+		"start",
 	}
-
-	// FlagD is the default provider name externally
-	if featureFlag.Spec.ServiceProvider != nil && featureFlag.Spec.ServiceProvider.Name != "flagd" {
-		commandSequence = append(commandSequence, "--service-provider")
-		commandSequence = append(commandSequence, "http")
-	}
-
+	var envs []corev1.EnvVar
 	var volumeMounts []corev1.VolumeMount
-	// Adds the sync provider if it is set
-	if featureFlag.Spec.SyncProvider != nil && !featureFlag.Spec.SyncProvider.IsKubernetes() {
-		commandSequence = append(commandSequence, "--sync-provider")
-		commandSequence = append(commandSequence, featureFlag.Spec.SyncProvider.Name)
-		// inject config map as volume if sync provider not kubernetes
-		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
-			Name: "flagd-config",
-			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: configMap,
+
+	for _, featureFlag := range featureFlags {
+		if featureFlag.Spec.FlagDSpec != nil {
+			if featureFlag.Spec.FlagDSpec.MetricsPort != 0 {
+				flagdMetricsPort = featureFlag.Spec.FlagDSpec.MetricsPort
+			}
+			envs = append(envs, featureFlag.Spec.FlagDSpec.Envs...)
+		}
+		switch {
+		// kubernetes sync is the default state
+		case featureFlag.Spec.SyncProvider == nil || featureFlag.Spec.SyncProvider.IsKubernetes():
+			fmt.Printf("FeatureFlagConfiguration %s using kubernetes sync implementation\n", featureFlag.Name)
+			commandSequence = append(
+				commandSequence,
+				"--uri",
+				fmt.Sprintf(
+					"core.openfeature.dev/%s/%s",
+					featureFlag.ObjectMeta.Namespace,
+					featureFlag.ObjectMeta.Name,
+				),
+			)
+			// if http is explicitly set
+		case featureFlag.Spec.SyncProvider.IsHttp():
+			fmt.Printf("FeatureFlagConfiguration %s using http sync implementation\n", featureFlag.Name)
+			if featureFlag.Spec.SyncProvider.HttpSyncConfiguration != nil {
+				commandSequence = append(
+					commandSequence,
+					"--uri",
+					featureFlag.Spec.SyncProvider.HttpSyncConfiguration.Target,
+					"--bearer-token",
```

Comment:
Is bearer-token now mandatory or are we passing a nil value here?

Reply:
I've now wrapped this in an `if`